packages/manager/apps/web/client/app/domain/zone/product-zone-activate/activate.controller.js

```diff
@@ -0,0 +1,50 @@
+import filter from 'lodash/filter';
+
+export default class {
+  /* @ngInject */
+  constructor($filter, $timeout) {
+    this.$filter = $filter;
+    this.$timeout = $timeout;
+  }
+
+  $onInit() {
+    // Auto-select duration
+    this.prices = filter(this.serviceOption.prices, price => price.pricingMode === 'default');
+    [this.price] = this.prices;
+    this.interval = this.price.interval;
+    this.isEditable = true;
+    if (this.prices.length === 1) {
+      // Go directly to the next step
+      this.currentIndex = 1;
+      this.isEditable = false;
+    }
+  }
+
+  resetCart() {
+    if (this.cart) {
+      this.cart = undefined;
+      this.cartId = undefined;
+    }
+  }
+
+  async prepareCheckout() {
+    if (!this.cart && !this.checkoutLoading) {
+      this.checkoutLoading = true;
+      const { cart, cartId } = await this.prepareOrderCart(this.price);
+      this.$timeout(() => {
```

Comment:
I know the `async/await` syntax is sometimes easier to understand but then we have to `$timeout` to follow digest cycle and I'm not sure it is better. It is ok here because we're in a controller so there is smaller chances of reuse than for a `service`